app/src/main/java/com/andrew/tasky/presentation/agenda/AgendaViewModel.kt

```diff
@@ -31,11 +32,47 @@ class AgendaViewModel@Inject constructor(
 
     fun setDateSelected(dateUserSelected: LocalDate) {
         _dateSelected.value = dateUserSelected
+        _calendarDateItemList.value.map { item ->
+            item.isSelected = dateUserSelected == item.date
+        }
```

Comment:
Here, you're not changing the actual state since you don't do anything with the result. To assign a new state, you have to set it to `_calendarDateItemList` like this:
```
_calendarDateItemList.value = _calendarDateItemList.value.map { ... }
```

This causes the state to not be changed which therefore doesn't trigger your collector in the UI and therefore doesn't properly update your background in the date list.